### PR TITLE
FPGA: Add `device_global` sample

### PR DESCRIPTION
**Pull request by artemrad**
# Adding a New Sample(s)

## Description

Showcasing the `device_global` capabilities that users can explore in 2023.1 release

## Checklist
Administrative
- [x] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): @yuguen-intel
- [ N/A ] If you have any new dependencies/binaries, inform the oneAPI Code Samples Project Manager

Code Development
- [x] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [x] Adhere to readme template 
- [x] Enforce format via clang-format config file
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [x] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [x] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [ ] OSPDT Approval (see Project Manager for assistance)
- [ ] Compile using the following compiler flags and fix any warnings, the falgs are: "/Wall -Wformat-security -Werror=format-security"
- [ N/A ] Bandit Scans (Python only)
- [ ] Virus scan

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Project Manager
- [ ] Tested using Dev Cloud when applicable

### CODE REVIEW

**Comment by artemrad** [comment]
Linux: https://spetc.intel.com/testsummary?trview=0&testRunIds=7435850

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
+# `device_global`
+This FPGA tutorial explains how to use `device_global` class as a way of keeping a state between multiple invocations of a kernel.
+
+| Optimized for                     | Description
+|:---                                 |:---
+| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| Hardware                          | Intel&reg; Programmable Acceleration Card (PAC) with Intel Arria&reg; 10 GX FPGA; <br> Intel&reg; FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix&reg; 10 SX) <br> Intel&reg; FPGA 3rd party / custom platforms with oneAPI support <br> **Note**: Intel&reg; FPGA PAC hardware is only compatible with Ubuntu 18.04*
+| Software                          | Intel&reg; oneAPI DPC++ Compiler <br> Intel&reg; FPGA Add-On for oneAPI Base Toolkit
+| What you will learn               | The basic usage of the `device_global` class <br> How to initialize a `device_global` to non-zero values<br>
+| Time to complete                  | 15 minutes
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:10]: This part has been changed in the latest code samples to split this table in two and add a "prerequisite" section.
See https://github.com/oneapi-src/oneAPI-samples/tree/development/DirectProgramming/C%2B%2BSYCL_FPGA/Tutorials/DesignPatterns/autorun for example.

To fill this prerequisite section correctly, you need to register your new sample in the `/DirectProgramming/C++SYCL_FPGA/README.md` file

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    });
+  });
+}
+```
+
+### Additional Documentation
+- [SYCL `device_global` Language Specification](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_device_global.asciidoc) helps understand the API and of `device_globals` and restrictions on it's usage
+- [Explore SYCL* Through Intel&reg; FPGA Code Samples](https://software.intel.com/content/www/us/en/develop/articles/explore-dpcpp-through-intel-fpga-code-samples.html) helps you to navigate the samples and build your knowledge of FPGAs and SYCL.
+- [FPGA Optimization Guide for Intel&reg; oneAPI Toolkits](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide) helps you understand how to target FPGAs using SYCL and Intel&reg; oneAPI Toolkits.
+- [Intel&reg; oneAPI Programming Guide](https://software.intel.com/en-us/oneapi-programming-guide) helps you understand target-independent, SYCL-compliant programming using Intel&reg; oneAPI Toolkits.
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:51]: Please remove this section

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+> the root of your oneAPI installation.
+>
+> Linux*:
+> - For system wide installations: `. /opt/intel/oneapi/setvars.sh`
+> - For private installations: `. ~/intel/oneapi/setvars.sh`
+>
+> Windows*:
+> - `C:\Program Files(x86)\Intel\oneAPI\setvars.bat`
+>
+>For more information on environment variables, see **Use the setvars Script** for [Linux or macOS](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-linux-or-macos.html), or [Windows](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-windows.html).
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:70]:
```suggestion
> **Note**: When working with the command-line interface (CLI), you should configure the oneAPI toolkits using environment variables. 
> Set up your CLI environment by sourcing the `setvars` script located in the root of your oneAPI installation every time you open a new terminal window. 
> This practice ensures that your compiler, libraries, and tools are ready for development.
>
> Linux*:
> - For system wide installations: `. /opt/intel/oneapi/setvars.sh`
> - For private installations: ` . ~/intel/oneapi/setvars.sh`
> - For non-POSIX shells, like csh, use the following command: `bash -c 'source <install-dir>/setvars.sh ; exec csh'`
>
> Windows*:
> - `C:\Program Files(x86)\Intel\oneAPI\setvars.bat`
> - Windows PowerShell*, use the following command: `cmd.exe "/K" '"C:\Program Files (x86)\Intel\oneAPI\setvars.bat" && powershell'`
>
> For more information on configuring environment variables, see [Use the setvars Script with Linux* or macOS*](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-linux-or-macos.html) or [Use the setvars Script with Windows*](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-windows.html).
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+The basic steps to build and run a sample using VS Code include:
+ - Download a sample using the extension **Code Sample Browser for Intel&reg; oneAPI Toolkits**.
+ - Configure the oneAPI environment with the extension **Environment Configurator for Intel&reg; oneAPI Toolkits**.
+ - Open a Terminal in VS Code (**Terminal>New Terminal**).
+ - Run the sample in the VS Code terminal using the instructions below.
+ - (Linux only) Debug your GPU application with GDB for Intel&reg; oneAPI toolkits using the **Generate Launch Configurations** extension.
+
+To learn more about the extensions, see the 
+[Using Visual Studio Code with Intel&reg; oneAPI Toolkits User Guide](https://www.intel.com/content/www/us/en/develop/documentation/using-vs-code-with-intel-oneapi/top.html).
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:92]: Remove these sections

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md (hunk 3 above)
Comment (yuguen-intel) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:70]: Please use this new template for the CLI env variable

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+the `VERBOSE=1` argument:
+``make VERBOSE=1``
+For more comprehensive troubleshooting, use the Diagnostics Utility for
+Intel&reg; oneAPI Toolkits, which provides system checks to find missing
+dependencies and permissions errors.
+[Learn more](https://www.intel.com/content/www/us/en/develop/documentation/diagnostic-utility-user-guide/top.html).
+
+### In Third-Party Integrated Development Environments (IDEs)
+
+You can compile and run this tutorial in the Eclipse* IDE (in Linux*) and the Visual Studio* IDE (in Windows*). For instructions, refer to the following link: [FPGA Workflows on Third-Party IDEs for Intel&reg; oneAPI Toolkits](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-oneapi-dpcpp-fpga-workflow-on-ide.html).
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:189]: Please remove these sections

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+ 1. Run the sample on the FPGA emulator (the kernel executes on the CPU):
+     ```
+     ./device_global.fpga_emu     (Linux)
+     device_global.fpga_emu.exe   (Windows)
+     ```
+2. Run the sample on the FPGA simulator device:
+     ```
+     ./device_global.fpga_sim     (Linux)
+     device_global.fpga_sim.exe   (Windows)
+     ```
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:202]:
```suggestion
    * On Linux
        ```bash
        CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=1 ./device_global.fpga_sim
        ```
    * On Windows
        ```bash
        set CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=1
        device_global.fpga_sim.exe
        set CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=
        ```
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/sample.json [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+          "make report"
+        ]
+      }
+    ],
+    "windows": [
+      {
+        "id": "fpga_emu",
+        "steps": [
+          "icpx --version",
+          "cd ../../..",
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/sample.json:49]:
```suggestion
          "cd ../../../..",
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/sample.json [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+          "cmake -G \"NMake Makefiles\" ../Tutorials/Features/experimental/device_global",
+          "nmake fpga_emu",
+          "device_global.fpga_emu.exe"
+        ]
+      },
+      {
+        "id": "report",
+        "steps": [
+          "icpx --version",
+          "cd ../../..",
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/sample.json:61]:
```suggestion
          "cd ../../../..",
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/device_global.cpp [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+#if FPGA_SIMULATOR
+    auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+    auto selector = sycl::ext::intel::fpga_selector_v;
+#else // #if FPGA_EMULATOR
+    auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+    sycl::queue q(selector, fpga_tools::exception_handler,
+                  sycl::property::queue::enable_profiling{});
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/device_global.cpp:72]:
```suggestion
    sycl::queue q(selector, fpga_tools::exception_handler,
                  sycl::property::queue::enable_profiling{});

    auto device = q.get_device();

    std::cout << "Running on device: "
              << device.get_info<sycl::info::device::name>().c_str()
              << std::endl;                  
                  
```

**Review by yuguen-intel — CHANGES_REQUESTED** [review]
A few modifications to match the current samples expectations

File: DirectProgramming/C++SYCL_FPGA/README.md [hunk 10]
```diff
@@ -78,6 +78,7 @@ flowchart LR
 |:---                                                                       |:---                                                  |:---
 | [ac_fixed](Tutorials/Features/ac_fixed)                                   | [Tutorials/Features](Tutorials/Features)             | How different methods of `ac_fixed` number construction affect hardware resource utilization <br> Recommended method for constructing `ac_fixed` numbers in your kernel <br> Accessing and using the `ac_fixed` math library functions <br> Trading off accuracy of results for reduced resource usage on the FPGA
 | [ac_int](Tutorials/Features/ac_int)                                       | [Tutorials/Features](Tutorials/Features)             | Using the `ac_int` data type for basic operations <br> Efficiently using the left shift operation <br> Setting and reading certain bits of an `ac_int` number
+| [device_global (experimental)](Tutorials/Features/experimental/device_global)| [Tutorials/Features](Tutorials/Features)             | The basic usage of the `device_global` class <br> How to initialize a `device_global` to non-zero values<br>
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/README.md:81]:
```suggestion
| [device_global (experimental)](Tutorials/Features/experimental/device_global)| [Tutorials/Features](Tutorials/Features)             | The basic usage of the `device_global` class <br> How to initialize a `device_global` to non-zero values

```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md [hunk 11]
```diff
@@ -0,0 +1,225 @@
+# `device_global`
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:1]:
```suggestion
# `device_global` Sample
```

**Review by yuguen-intel — CHANGES_REQUESTED** [review]

**Review by yuguen-intel — APPROVED** [review]

**Review by tomlenth — APPROVED** [review]
Thanks, super thorough

**Review by jkinsky — APPROVED** [review]
There are some issues to correct, but overall this looks good for the first iteration.

**Comment by yuguen-intel** [comment]
Please incorporate these changes in your PR: https://github.com/oneapi-src/oneAPI-samples/pull/1307

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt [hunk 12]
```diff
@@ -6,12 +6,22 @@ set(FPGA_TARGET ${TARGET_NAME}.fpga)
...
 # FPGA board selection
 if(NOT DEFINED FPGA_DEVICE)
-    set(FPGA_DEVICE "intel_a10gx_pac:pac_a10")
+    set(FPGA_DEVICE "Agilex")
     message(STATUS "FPGA_DEVICE was not specified.\
-                    \nConfiguring the design to run on the default FPGA board ${FPGA_DEVICE} (Intel(R) PAC with Intel Arria(R) 10 GX FPGA). \
-                    \nPlease refer to the README for information on board selection.")
+            \nConfiguring the design to the default FPGA family: ${FPGA_DEVICE}\
+            \nPlease refer to the README for information on target selection.")
+    set(IS_BSP "0")
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt:13]:
```suggestion
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt [hunk 13]
```diff
@@ -6,12 +6,22 @@ set(FPGA_TARGET ${TARGET_NAME}.fpga)
...
 else()
-    message(STATUS "Configuring the design to run on FPGA board ${FPGA_DEVICE}")
+    message(STATUS "Configuring the design with the following target: ${FPGA_DEVICE}")
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt:15]:
```suggestion
    string(TOLOWER ${FPGA_DEVICE} FPGA_DEVICE_NAME)
    if(FPGA_DEVICE_NAME MATCHES ".*a10.*" OR FPGA_DEVICE_NAME MATCHES ".*arria10.*")
      set(DEVICE_FLAG "A10")
    elseif(FPGA_DEVICE_NAME MATCHES ".*s10.*" OR FPGA_DEVICE_NAME MATCHES ".*stratix10.*")
      set(DEVICE_FLAG "S10")
    elseif(FPGA_DEVICE_NAME MATCHES ".*agilex.*")
      set(DEVICE_FLAG "Agilex")
    endif()
    message(STATUS "Configuring the design with the following target: ${FPGA_DEVICE}")
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt [hunk 14]
```diff
@@ -6,12 +6,22 @@ set(FPGA_TARGET ${TARGET_NAME}.fpga)
...
+
+    # Check if the target is a BSP
+    if(IS_BSP MATCHES "1" OR FPGA_DEVICE MATCHES ".*pac_a10.*|.*pac_s10.*")
+        set(IS_BSP "1")
+    else()
+        set(IS_BSP "0")
+        message(STATUS "The selected target ${FPGA_DEVICE} is assumed to be an FPGA part number, so USM will be enabled by default.")
+        message(STATUS "If the target is actually a BSP that does not support USM, run cmake with -DIS_BSP=1.")
+    endif()
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt:24]:
```suggestion
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt [hunk 15]
```diff
@@ -6,12 +6,22 @@ set(FPGA_TARGET ${TARGET_NAME}.fpga)
...
 endif()
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt:26]:
```suggestion

if(NOT DEFINED DEVICE_FLAG)
    message(FATAL_ERROR "An unrecognized or custom board was passed, but DEVICE_FLAG was not specified. \
                         Make sure you have set -DDEVICE_FLAG=A10, -DDEVICE_FLAG=S10 or -DDEVICE_FLAG=Agilex.")
endif()

```

Comment (yuguen-intel) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt:26]: And 
`if(FPGA_DEVICE MATCHES ".*a10.*")`
need to be modified to 
`if(DEVICE_FLAG MATCHES "A10")`

**Review by yuguen-intel — CHANGES_REQUESTED** [review]

**Review by pmpeter1 — APPROVED** [review]

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+> - Questa*-Intel® FPGA Edition
+> - Questa*-Intel® FPGA Starter Edition
+> - ModelSim® SE
+>
+> When using the hardware compile flow, Intel® Quartus® Prime Pro Edition must be installed and accessible through your PATH.
+>
+> :warning: Make sure you add the device files associated with the FPGA that you are targeting to your Intel® Quartus® Prime installation.
+
+>**Note**: Intel® FPGA PAC hardware is only compatible with Ubuntu 18.04*.
+
```
Comment (artemrad) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:50]:
```suggestion
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+     ```
+   * Compile for simulation (fast compile time, targets simulated FPGA device, reduced data size):
+     ```
+     make fpga_sim
+     ```
+   * Compile for FPGA hardware (longer compile time, targets FPGA device):
+     ```
+     make fpga
+     ```
+3. (Optional) As the FPGA hardware compile may take several hours to complete, FPGA precompiled binaries (compatible with Linux* Ubuntu* 18.04) can be downloaded <a href="https://iotdk.intel.com/fpga-precompiled-binaries/latest/device_global.fpga.tar.gz" download>here</a>.
```
Comment (artemrad) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/README.md:151]:
```suggestion
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+endif()
+
+# This is a Windows-specific flag that enables exception handling in host code
+if(WIN32)
+    set(WIN_FLAG "/EHsc")
+endif()
+
+# Hyper-optimization is turned off to allow the same set of latency control parameters to work on both A10 and S10.
+# Users can turn it back on and try their own parameters on S10.
+if(FPGA_DEVICE MATCHES ".*a10.*")
```
Comment (artemrad) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt:38]:
```suggestion
if(FPGA_DEVICE MATCHES ".*A10.*")
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+endif()
+
+# This is a Windows-specific flag that enables exception handling in host code
+if(WIN32)
+    set(WIN_FLAG "/EHsc")
+endif()
+
+# Hyper-optimization is turned off to allow the same set of latency control parameters to work on both A10 and S10.
+# Users can turn it back on and try their own parameters on S10.
+if(FPGA_DEVICE MATCHES ".*A10.*")
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt:38]:
```suggestion
if(DEVICE_FLAG MATCHES "A10")
```

**Review by yuguen-intel — CHANGES_REQUESTED** [review]

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+if(NOT DEFINED FPGA_DEVICE)
+    set(FPGA_DEVICE "Agilex")
+    message(STATUS "FPGA_DEVICE was not specified.\
+            \nConfiguring the design to the default FPGA family: ${FPGA_DEVICE}\
+            \nPlease refer to the README for information on target selection.")
+else()
+    string(TOLOWER ${FPGA_DEVICE} FPGA_DEVICE_NAME)
+    if(FPGA_DEVICE_NAME MATCHES ".*A10.*" OR FPGA_DEVICE_NAME MATCHES ".*arria10.*")
+      set(DEVICE_FLAG "A10")
+    elseif(FPGA_DEVICE_NAME MATCHES ".*S10.*" OR FPGA_DEVICE_NAME MATCHES ".*stratix10.*")
```
Comment (artemrad) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/device_global/src/CMakeLists.txt:17]:
```suggestion
    if(FPGA_DEVICE_NAME MATCHES ".*a10.*" OR FPGA_DEVICE_NAME MATCHES ".*arria10.*")
      set(DEVICE_FLAG "A10")
    elseif(FPGA_DEVICE_NAME MATCHES ".*a10.*" OR FPGA_DEVICE_NAME MATCHES ".*stratix10.*")
```

**Review by yuguen-intel — APPROVED** [review]

**Pull request merged**